superset/assets/src/dashboard/util/publishSubscriberUtil.js

```diff
@@ -30,17 +32,76 @@ export function getUniqueActionsForSlice(slice) {
 export function getModalSliceIDFor(publishSubscriberMap, publisherId) {
     if (publishSubscriberMap && publishSubscriberMap.hasOwnProperty('subscribers')) {
         const subscribers = Object.values(publishSubscriberMap.subscribers);
-        var item = _.find(subscribers, function (item) {
+        var item = find(subscribers, function (item) {
             return ((item.actions.indexOf(USE_AS_MODAL) > -1) && (Object.keys(item.linked_slices).indexOf(publisherId) != -1));
         });
         return item ? item.id : undefined;
     }
     return undefined
 }
 
+export function getSubHeaderForSlice(subscribers, chartId, filters) {
+    let subHeader = '';
+    if (chartId != -1 && subscribers && subscribers[chartId]) {
+        const subscriber = subscribers[chartId];
+        if (subscriber.actions.indexOf(INCLUDE_IN_TITLE) > -1) {
+            subHeader = '';
+            for (let lsKey in subscriber.linked_slices) {
+                if (keyExists(lsKey, filters)) {
+                    subscriber.linked_slices[lsKey].forEach(linkedSlice => {
+                        if (linkedSlice.actions.indexOf(INCLUDE_IN_TITLE) > -1) {
+                            let columnName = linkedSlice['col'];
+                            let subHeaderValues = filters[lsKey][columnName];
+                            const values = getSubHeaderValues(subHeaderValues);
+
+                            if (subHeaderValues) {
+                                (subHeader != '') ? values.push(subHeader) : [];
+                                subHeader = values.join(' | ');
+                            }
+                        }
+                    });
+                }
+            }
+        }
+    }
+    return subHeader;
+}
+
+function getSubHeaderValues(headerValues) {
+    let subTitleHeaders = '';
+    if (headerValues) {
+        if (headerValues.constructor == Array) {
+            subTitleHeaders = headerValues.concat();
+        }
+        else {
+            subTitleHeaders = [headerValues];
+        }
+    }
+    else {
+        subTitleHeaders = [];
+    }
+
+    return subTitleHeaders;
+}
+
+export function keyExists(key, search) {
+    if (!search || (search.constructor !== Array && search.constructor !== Object)) {
+        return false;
+    }
+
+    for (let i = 0; i < search.length; i++) {
```

Comment:
if search is object, in that case search.length will not work.

Reply:
It will automatically get out and go to the last line to return key in object.